openpgp/keys.go

```diff
@@ -421,7 +421,7 @@ EachPacket:
 			// Section 5.2.3.3, if there are several self-signatures,
 			// we should take the newer one.
 			if current != nil &&
-				(current.SelfSignature == nil || pkt.CreationTime.After(current.SelfSignature.CreationTime)) &&
+				(current.SelfSignature == nil || !pkt.CreationTime.Before(current.SelfSignature.CreationTime)) &&
```

Comment:
Could we add a comment here mentioning that this otherwise-innocuous-looking check is actually really subtle?